### ISSUE TITLE
Nix injection incorrectly highlights elements as strings

### ISSUE THREAD

**Issue by mydumpfire** [cstrahan/tree-sitter-nix#49]
This is how bash is normally highlighted (on helix, with the built-in `tokyonight` theme):

![image](https://github.com/nix-community/tree-sitter-nix/assets/134539278/4881f5e9-b0c3-46c9-a42d-2bfe78a9668a)

And this is how it looks like in an injected nix string block:

![image](https://github.com/nix-community/tree-sitter-nix/assets/134539278/abd664dc-4fbb-4b4c-b043-2778da05e54f)

A lot of stuff that shouldn't be highlighted as a string, gets highlighted as one.

On the other hand, neovim doesn't seem to have this problem:

![image](https://github.com/nix-community/tree-sitter-nix/assets/134539278/6555e3a9-8a88-49a0-b6b3-0b794d520582)

Anyone know what's causing this issue?

**Comment by asymmetric** [comment]
Are you sure you're using tree-sitter-nix in both editors? Generally, it would be good to post minimal configurations so that people can reproduce.

**Comment by mydumpfire** [comment]
Yes, they're both using treesitter as the syntax highlighter. I can use `:Inspect` and `:InspectTree` on the neovim buffer and view the structure:

![image](https://github.com/nix-community/tree-sitter-nix/assets/134539278/05c669d6-6f87-4ea5-9c61-a1fc5c4ef465)

![image](https://github.com/nix-community/tree-sitter-nix/assets/134539278/8c7b46de-9d01-4644-abd3-0963de1bd195)

For helix, you don't need a config. You should be able to reproduce it with the defaults: `hx -c ''`.

For neovim, you just need to have `nvim-treesitter` installed with the `nix` parser. If you use `lazy.nvim` as a package manager you can use:

```lua
require("lazy").setup({{
    "nvim-treesitter/nvim-treesitter",
    build = ":TSUpdate",
    config = function () 
      local configs = require("nvim-treesitter.configs")

      configs.setup({
          ensure_installed = { "nix" },
          sync_install = false,
          highlight = { enable = true },
          indent = { enable = true },  
        })
    end
 }})
```

For other installation methods: https://github.com/nvim-treesitter/nvim-treesitter/wiki/Installation

**Comment by the-mikedavis** [comment]
This is because nvim-treesitter highlights `(command argument: (word))`: https://github.com/nvim-treesitter/nvim-treesitter/blob/8a9bef9327bc06c9c4158d4873d87bb1bf3ed112/queries/bash/highlights.scm#L172-L175

Helix doesn't highlight the `(word)` nodes so they keep the string highlight.
